### ISSUE TITLE
Loss increases during second fine-tuning phase

### ISSUE THREAD

**Issue by jackatls** [stanford-futuredata/ColBERT#385]
This is the result after my first fine-tuning.
![92544a92701a478d1f74ae22f7859cb7](https://github.com/user-attachments/assets/326a640d-aeb9-4fb1-9b8c-e0ee1fa9f860)
This is the second fine-tuning, where the checkpoint saved after the first fine-tuning was loaded. The loss starts lower than the loss value from the first fine-tuning but gradually increases.
![4f402db7f7fd84853bbce42d0ebe01d2](https://github.com/user-attachments/assets/62298064-9b31-4f79-8173-2bb2a75650e2)
Could it be an issue with the loading process, such as certain optimizer states not being properly restored? I load the saved checkpoint by
`trainer.train(checkpoint=lastCheckpointDir)`

I would greatly appreciate your help!
